src/ext/java/org/opentripplanner/ext/emissions/Co2EmissionsDataReader.java

```diff
@@ -0,0 +1,113 @@
+package org.opentripplanner.ext.emissions;
+
+import com.csvreader.CsvReader;
+import com.esotericsoftware.minlog.Log;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.zip.ZipFile;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Co2EmissionsDataReader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(Co2EmissionsDataReader.class);
+  private Map<FeedScopedId, Double> emissionsData = new HashMap<>();
+
+  public Map<FeedScopedId, Double> readGtfs(String filePath) {
+    try {
+      InputStream feedInfoStream = new FileInputStream(filePath + "/feed_info.txt");
```

Comment:
Please use the try-with-resources idiom when dealing with InputStreams: https://www.programiz.com/java-programming/try-with-resources